packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Watch,
+  Prop,
+  State,
+  h,
+} from '@stencil/core';
+import { assignLanguage, observerConfig } from '../../utils/utils';
+import { inheritAttributes } from '../../utils/utils';
+import i18n from './i18n/i18n';
+
+@Component({
+  tag: 'gcds-link',
+  styleUrl: 'gcds-link.css',
+  shadow: true,
+})
+export class GcdsLink {
+  @Element() el: HTMLElement;
+
+  private shadowElement?: HTMLElement;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Set the link size
+   */
+  @Prop({ mutable: true }) size?: 'regular' | 'small' | 'inherit' = 'inherit';
+  @Watch('size')
+  validateSize(newValue: string) {
+    const values = ['regular', 'small', 'inherit'];
+
+    if (!values.includes(newValue)) {
+      this.size = 'inherit';
+    }
+  }
+
+  /**
+   * Sets the display behavior of the link
+   */
+  @Prop() display?: 'block' | 'inline' | 'inline-block' = 'inline';
```

Comment:
Sorry, I'm gonna be annoying again :D I just remembered that CSS has some weird specs where it doesn't actually show the `text-decoration` for inline blocks. Since we want to use underlines for links to clearly mark them as links, let's remove the `inline-block` option everywhere, please.

Reply:
Not annoying; good catch!